src/migrations/20240520000000-merge_duplicate_args.js

```diff
@@ -0,0 +1,232 @@
+            argid,
+            argfrid
+          FROM argfr_on_donor_args aoda
+          WHERE choice_rank = 1
+            AND argid != target_arg
+        )
+        UPDATE "ActivityReportGoalFieldResponses" AS argfr
+        SET "activityReportGoalId" = target_arg
+        FROM arg_merges am
+        JOIN unmatched u
+          ON u.donor_arg = am.donor_arg
+        WHERE argfr.id = u.argfrid
+        RETURNING
+          id argfrid,
+          am.donor_arg original_arg
+      ) SELECT * FROM updater
+      ;
+
+      -- Delete duplicate objective ARGFRs
+      DROP TABLE IF EXISTS deleted_argfrs;
+      CREATE TEMP TABLE deleted_argfrs
+      AS
+      WITH updater AS (
+        DELETE FROM "ActivityReportGoalFieldResponses"
+        USING arg_merges
+        WHERE "activityReportGoalId" = donor_arg
+        RETURNING
+          id argfrid,
+          donor_arg
+      ) SELECT * FROM updater
+      ;
+
+      -- Relink any ActivityReportGoalResources connected to the
+      -- duplicate (and therefore donor) ARG
+      -- Because there could theoretically be multiple resources on
+      -- multiple duplicates, we need to rank the ARGRs referring to
+      -- a particular resource-goal pair and select just one of each.
+      -- There's one target_arg per goal so we use that as a proxy.
+      -- Just for simplicity, ARGRs that are already on the target
+      -- ARG are left alone and the corresponding responses on donors
+      -- will be deleted.
+      --
+      -- At time of writing this is all theoretical as there aren't
+      -- any ARGRs at all but this could change by the time it runs
+      DROP TABLE IF EXISTS relinked_argrs;
+      CREATE TEMP TABLE relinked_argrs
+      AS
+      WITH updater AS (
+        WITH argr_on_donor_args AS (
+          SELECT
+            donor_arg,
+            target_arg,
+            argr."activityReportGoalId" argid,
+            argr."resourceId" resourceid,
+            argr.id argrid,
+            ROW_NUMBER() OVER (
+              PARTITION BY target_arg, argr."resourceId"
+              ORDER BY argr."activityReportGoalId" = target_arg, argr."updatedAt" DESC, argr.id
+            ) choice_rank
+          FROM arg_merges am
+          JOIN "ActivityReportGoalResources" argr
+            ON am.donor_arg = argr."activityReportGoalId"
+        ), unmatched AS (
+          SELECT
+            donor_arg,
+            argid,
+            argrid
+          FROM argr_on_donor_args aoda
+          WHERE choice_rank = 1
+            AND argid != target_arg
+        )
+        UPDATE "ActivityReportGoalResources" AS argr
+        SET "activityReportGoalId" = target_arg
+        FROM arg_merges am
+        JOIN unmatched u
+          ON u.donor_arg = am.donor_arg
+        WHERE argr.id = u.argrid
+        RETURNING
+          id argrid,
+          am.donor_arg original_arg
+      ) SELECT * FROM updater
+      ;
+      -- Delete duplicate objective ARGRs
+      DROP TABLE IF EXISTS deleted_argrs;
```

Comment:
Would also do a quick double check here inserting duplicate resource records for the same ar and goal, just making sure we keep one of them.